### DISCUSSION TITLE
Resistor Selection

### DISCUSSION THREAD

**Discussion by AKMedi9** [taligentx/dscKeybusInterface#262] [dscKeybusInterface Discussion] [answered]
The documentation suggests using a 33k resistor for the clock and read pins for esp8266/esp32 and a 15k resistor for arduino boards. The PCB schematic lists 15k resistor for the NodeMCU board. Is this resistor value critical? Im building a board with an esp32 NodeMCU and I was going to use a 33k resistor here but I just wanted to double check.

![Screen Shot 2021-12-16 at 6 33 56 PM](https://user-images.githubusercontent.com/96084838/146469597-7b4f0b3c-6993-4d48-9b29-c78eccec65b6.png)
![Screen Shot 2021-12-16 at 6 35 41 PM](https://user-images.githubusercontent.com/96084838/146469611-fc86ea41-c136-41d5-a3a6-baf7d9c5ea4d.png)

**Comment by kghandi** [comment, marked as answer]
DSC operates at 12.6V, Arduino at 5V, and ESP at 3.3V. 
The resistors act as a voltage divider. 
Vout = Vin R2 / (R1+R2).  
So with 10k and 15k we get:
Vout = 12.6 * 10 / 25 = 5.06 good for Arduino
And with 10k and 33k:
Vout = 12.6 * 10 / 43 = 2.93 good for ESP

**Comment by AKMedi9** [comment]
Thanks that answers my question.

**Comment by realthk** [comment]
Or rather build an isolated ground version
[https://github.com/Dilbert66/esphome-dsckeybus](https://github.com/Dilbert66/esphome-dsckeybus)

It is easy to do even on a prototype board with all straight wires
![image](https://user-images.githubusercontent.com/5654575/152696090-f3426783-33be-40e9-87e5-76b56046f0d6.png)

This is safer and won't does not put load on the DSC Aux line.
